controller/array_action/array_mediator_xiv.py

```diff
@@ -261,21 +262,23 @@ def get_snapshot(self, snapshot_name, pool_id=None):
         if not cli_snapshot:
             return None
         if not cli_snapshot.master_name:
-            raise controller_errors.SnapshotNameBelongsToVolumeError(cli_snapshot.name, self.endpoint)
+            raise controller_errors.ExpectedSnapshotButFoundVolumeError(cli_snapshot.name, self.endpoint)
         array_snapshot = self._generate_snapshot_response(cli_snapshot)
         return array_snapshot
 
-    def get_snapshot_by_id(self, snapshot_id):
+    def get_object_by_id(self, object_id, object_type):
         try:
-            cli_snapshot = self.client.cmd.vol_list(wwn=snapshot_id).as_single_element
+            cli_object = self.client.cmd.vol_list(wwn=object_id).as_single_element
         except xcli_errors.IllegalValueForArgumentError as ex:
             logger.exception(ex)
             raise controller_errors.IllegalObjectID(ex.status)
-        if not cli_snapshot:
+        if not cli_object:
             return None
-        if not cli_snapshot.master_name:
-            raise controller_errors.SnapshotIdBelongsToVolumeError(snapshot_id, self.endpoint)
-        return self._generate_snapshot_response(cli_snapshot)
+        if object_type is controller_config.SNAPSHOT_TYPE_NAME:
+            if not cli_object.master_name:
+                raise controller_errors.ExpectedSnapshotButFoundVolumeError(object_id, self.endpoint)
```

Comment:
as you can see, `get_snapshot` and `get_object_by_id` (formerly known as `get_snapshot_by_id`) are almost identical, because of the (now legacy) `Snapshot(Id|Name)BelongsToVolumeError`s.
so now they can finally be united (as part of [CSI-1506](https://github.com/IBM/ibm-block-csi-driver/pull/158#discussion_r428914250), [CSI-1802](https://github.com/IBM/ibm-block-csi-driver/pull/243), or even this PR)!
let me know in which of them you think it should be done

Reply:
only for xiv?

Reply:
these functions are also very similar in ds8k, but without a reason/ticket for it (though they are shorter than in xiv).

in svc, before this PR, `get_snapshot_by_id` called `get_snapshot` (so no duplication there), and in this PR currently they just seem different (so still no actual duplication).

if you think the interface can be further reduced to avoid these duplications, feel free to bring up suggestions.

Reply:
its relevant only on xiv.. only there there is no difference..

Reply:
CSI-2139